kiwi/bootloader/config/base.py

```diff
@@ -298,12 +298,15 @@ def get_boot_path(self, target='disk'):
                 filesystem = self.xml_state.build_type.get_filesystem()
                 volumes = self.xml_state.get_volumes()
                 if filesystem == 'btrfs' and volumes:
-                    # boot or boot/grub2 on a subvolume prevents grub from
-                    # finding its config file
+                    # grub boot data paths must not be in a subvolume
+                    # otherwise grub won't be able to find its config file
+                    grub_boot_data_paths = ['boot', 'boot/grub', 'boot/grub2']
```

Comment:
I'm not sure if this test is ordered in the way it is written, but if it is, you should always test for `boot/grub2` before `boot/grub`, because it's possible for distros to ship grub legacy and grub2 (Debian, Mageia, and a few others do).

I would also suggest naming all variables `grub2_<foo>` to make it clear grub2 is the only thing supported.